event/oracle/upgrade_oracle_event.go

```diff
@@ -0,0 +1,171 @@
+package oracle
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"github.com/edgelesssys/ego/enclave"
+	"github.com/medibloc/panacea-doracle/sgx"
+	"github.com/tendermint/tendermint/light/provider"
+
+	oracletypes "github.com/medibloc/panacea-core/v2/x/oracle/types"
+	"github.com/medibloc/panacea-doracle/event"
+	"github.com/medibloc/panacea-doracle/panacea"
+	log "github.com/sirupsen/logrus"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+)
+
+type UpgradeOracleEvent struct {
+	reactor event.Reactor
+}
+
+var _ event.Event = (*UpgradeOracleEvent)(nil)
+
+func NewUpgradeOracleEvent(s event.Reactor) UpgradeOracleEvent {
+	return UpgradeOracleEvent{s}
+}
+
+func (e UpgradeOracleEvent) GetEventType() string {
+	return "message"
+}
+
+func (e UpgradeOracleEvent) GetEventAttributeKey() string {
+	return "action"
+}
+
+func (e UpgradeOracleEvent) GetEventAttributeValue() string {
+	return "'OracleUpgrade'"
+}
+
+func (e UpgradeOracleEvent) EventHandler(event ctypes.ResultEvent) error {
+	uniqueID := event.Events[oracletypes.EventTypeUpgradeVote+"."+oracletypes.AttributeKeyUniqueID][0]
+	addressValue := event.Events[oracletypes.EventTypeUpgradeVote+"."+oracletypes.AttributeKeyOracleAddress][0]
+	queryClient := e.reactor.QueryClient()
+
+	oracleRegistration, err := queryClient.GetOracleRegistration(addressValue, uniqueID)
+	if err != nil {
+		log.Infof("failed to get oracleRegistration, voting ignored. uniqueID(%s), address(%s). %v", uniqueID, addressValue, err)
+		return err
+	}
```

Comment:
As done in #93 ([line](https://github.com/medibloc/panacea-doracle/pull/93/files#diff-fafe7cdb94cfba6f519545f3146bd5016f4ef6217768f2f97ae12355bda79d46R92)), error might be handled differently depending on the error is from doracle itself or not.
If the error is `NotFound`, doracle should vote `NO` I think.

Reply:
Well, if oracleRegistration is not found, we cannot vote it.
Because we vote for oracleRegistration.

If we can't find oracleRegistration we can't vote because we don't have anyone to vote for. :)

Reply:
like `dataSale`

Reply:
You're right. Thx for your works!